client/task-list/task-list.js

```diff
@@ -165,9 +170,57 @@ export const TaskList = ( {
 		} );
 	};
 
+	const remindTaskLater = ( { key, onDismiss } ) => {
+		createNotice(
+			'success',
+			__( 'Task postponed until tomorrow', 'woocommerce-admin' ),
+			{
+				actions: [
+					{
+						label: __( 'Undo', 'woocommerce-admin' ),
+						onClick: () => undoRemindTaskLater( key ),
+					},
+				],
+			}
+		);
+		const isCoreTaskList = name === 'task_list';
+		const taskListName = isCoreTaskList ? 'tasklist' : 'extended_tasklist';
+		recordEvent( `${ taskListName }_remindmelater_task`, {
+			task_name: key,
+		} );
+
+		const dismissTime = Date.now() + DAY_IN_MS;
+		updateOptions( {
+			woocommerce_task_list_remind_me_later_tasks: {
+				...remindMeLaterTasks,
+				[ key ]: dismissTime,
+			},
+		} );
+		if ( onDismiss ) {
+			onDismiss();
+		}
+	};
+
+	const undoRemindTaskLater = ( key ) => {
+		const {
+			// eslint-disable-next-line no-unused-vars
+			[ key ]: oldValue,
+			...updatedRemindMeLaterTasks
+		} = remindMeLaterTasks;
+
+		updateOptions( {
+			woocommerce_task_list_remind_me_later_tasks: updatedRemindMeLaterTasks,
+		} );
+	};
+
 	const getVisibleTasks = () => {
```

Comment:
I think this is complex enough that it deserves either a integration test (ie a test of the component's displaying of remindMeLaterTasks) or it needs to be refactored to take some of these things as args and unit tested. Thats just my gut feeling. This component has a lot of complexity in it.

Reply:
I added some extra test cases in this commit, specifically targeting the `getVisibleTasks` method - https://github.com/woocommerce/woocommerce-admin/pull/6923/commits/c44d79ac84a0400d13a7168c3ae7fdcc79b0ef42

I do agree that this component has a lot of complexity (too much I would say), some of the future work Jeff has been poking at, should help with extracting this I think.